### PR TITLE
Fixing lint and bugs in new tasks

### PR DESCRIPTION
**Pull request by csibbitt**
See https://github.com/infrawatch/osp-observability-ansible/pull/21#issuecomment-1341250833

### CODE REVIEW

File: playbooks/deploy_sgcore.yml [hunk 1]
```diff
@@ -5,18 +5,19 @@
...
     container_image: "{{ sgcore_image | default('quay.io/infrawatch/sg-core:latest') }}"
+    wdir: "{{ base_work_directory | default('/var/lib/osp-observability') }}"
   become: true
   become_method: sudo
   tasks:
     - name: Configure Ceilometer notification agent
       include_role:
         name: /usr/share/ansible/roles/osp_observability
         tasks_from: configure_ceilometer
-      vars:
```
Comment (csibbitt) [inline playbooks/deploy_sgcore.yml:15]:
```
Error was a <class 'ansible.errors.AnsibleError'>, original message: recursive loop detected in templ
ate string: {{ ceilometer_pipeline_path }}"}
```

File: playbooks/deploy_sgcore.yml [hunk 2]
```diff
@@ -5,18 +5,19 @@
     ceilometer_pipeline_path: "{{ ceilometer_pipeline_path | default('/etc/ceilometer/pipeline.yaml') }}"
     container_name: "{{ sgcore_container_name | default('sg-core') }}"
     container_image: "{{ sgcore_image | default('quay.io/infrawatch/sg-core:latest') }}"
+    wdir: "{{ base_work_directory | default('/var/lib/osp-observability') }}"
```
Comment (csibbitt) [inline playbooks/deploy_sgcore.yml:8]: Copied from prometheus playbook

File: roles/osp_observability/tasks/configure_ceilometer.yml [hunk 3]
```diff
@@ -1,8 +1,8 @@
 ---
```
Comment (csibbitt) [inline roles/osp_observability/tasks/configure_ceilometer.yml:1]: This was all from the linter

File: roles/osp_observability/tasks/configure_sgcore.yml [hunk 4]
```diff
@@ -1,13 +1,24 @@
...
     src: sgcore_config.yml.j2
     dest: "{{ workdir }}/sg-core/sgcore_config.yml"
+    owner: root
+    group: root
+    mode: '0644'
 
 - name: Spawn containerized sg-core with provided configuration
   include_role:
-    name: /usr/share/ansible/roles/spawn_container
+    name: spawn_container
```
Comment (csibbitt) [inline roles/osp_observability/tasks/configure_sgcore.yml:21]: Needed to live test in order to verify this

File: roles/osp_observability/tasks/configure_sgcore.yml [hunk 5]
```diff
@@ -1,13 +1,24 @@
 ---
 
+- name: Prepare configuration directory for sg-core
```
Comment (csibbitt) [inline roles/osp_observability/tasks/configure_sgcore.yml:3]:
```
fatal: [controller-0]: FAILED! => {"changed": false, "checksum": "6a3cc93acb47172d1059a94646d39e8f5ee1f6e0", "msg": "Destination directory /var/lib/osp-observability/sg-core does not exist"}
```

**Comment by csibbitt** [comment]
# Testing
Molecule testing would be nice but wouldn't actually prove these changes, so I did this instead for now...

## Undercloud Prepare
According to https://github.com/infrawatch/python-observabilityclient/blob/master/README.md#development and hints on https://github.com/infrawatch/osp-observability-ansible/pull/13

```
$ pwd
/home/stack/obsclient/osp-observability-ansible

$ ls -ld /usr/share/ansible/roles/spawn_container
lrwxrwxrwx. 1 root root 69 Dec  1 18:49 /usr/share/ansible/roles/spawn_container -> /home/stack/obsclient/osp-observability-ansible/roles/spawn_container

$ git reset --hard origin/csibbitt-fix-more-lint
HEAD is now at 51ea05e Fixing lint in new tasks

(undercloud) [stack@undercloud-0 osp-observability-ansible]$ git status
On branch csibbitt-fix-more-lint
Your branch is up to date with 'origin/csibbitt-fix-more-lint'.

nothing to commit, working tree clean

$ cat ~/test_params.yaml 
ceilometer_pipeline_path: "/var/lib/config-data/puppet-generated/ceilometer/etc/ceilometer/pipeline.yaml"
prometheus_remote_write:
[...]
```

## Converge 

```
$ ansible-playbook -i ~/overcloud-deploy/overcloud/tripleo-ansible-inventory.yaml -e @~/test_params.yaml playbooks/deploy_sgcore.yml 
[...]
TASK [spawn_container : Report result of container startup] **********************************************************************************************************************************
ok: [controller-0] => {                                                              
    "result": {                                     
        "actions": [                                                                                
            "pulled image quay.io/infrawatch/sg-core:latest",                                                                                                  
            "created sg-core"                                                                                                                                                                
        ],                              
[...]
                "CreateCommand": [          
                    "podman",                                                                                                                                                                 
                    "container",                                                                                                                                                             
                    "create",          
                    "--name",                               
                    "sg-core",
                    "--user",
                    "root",
                    "--restart=always",
                    "--network",
                    "host",
                    "--volume",
                    "/var/lib/osp-observability/sg-core/sgcore_config.yml:/etc/sg-core.conf.yaml:z",
                    "quay.io/infrawatch/sg-core:latest"
                ],
[...]
PLAY RECAP ***********************************************************************************************************************************************************************************
controller-0               : ok=17   changed=6    unreachable=0    failed=0    skipped=4    rescued=0    ignored=0   
```

## Verify
```
[root@controller-0 ~]# head -3 /var/lib/config-data/puppet-generated/ceilometer/etc/ceilometer/pipeline.yaml
---

sources:
[...]

[root@controller-0 ~]# systemctl status tripleo_ceilometer_agent_notification.service
● tripleo_ceilometer_agent_notification.service - ceilometer_agent_notification container
     Loaded: loaded (/etc/systemd/system/tripleo_ceilometer_agent_notification.service; enabled; vendor preset: disabled)
     Active: active (running) since Wed 2022-12-07 17:50:41 UTC; 3min 52s ago
[...]

[root@controller-0 ~]# head -3 /var/lib/osp-observability/sg-core/sgcore_config.yml 
---
plugindir: bin/
loglevel: debug


[root@controller-0 ~]# podman ps | grep sg-core
d010886f250d  quay.io/infrawatch/sg-core:latest                                                                                                                        7 minutes ago  Exited (0) Less than a second ago

[root@controller-0 ~]# podman logs sg-core | tail -1
2022-12-07 18:00:10 [ERROR] failed parsing config file [error: unmarshalling config yaml: yaml: line 9: mapping values are not allowed in this context]
```

## ~Can I still merge, please?~ Fixed!

~AFAICT this sg-core error is likely related to the fact that this hasn't merged: https://github.com/infrawatch/sg-core/pull/103, So the config file in this code is invalid. I was mainly verifying that I didn't obviously break anything, and that the lookup of the spawn_container role was working without the full path.~
(It turns out it was just a spacing issue)

File: roles/osp_observability/tasks/configure_sgcore.yml (hunk 4 above)
Comment (csibbitt) [reply roles/osp_observability/tasks/configure_sgcore.yml:21]: Originally a linter suggestion

**Comment by csibbitt** [comment]
The config file error was just a spacing issue.

```
[root@controller-0 ~]# podman logs sg-core | tail -4
2022-12-07 18:45:01 [ERROR] failed configuring transport [transport: socket, error: unable to determine socket type from configuration file. Should be either "unix" or "udp", received: tcp]
2022-12-07 18:45:01 [INFO] loaded application plugin [application: prometheus]
2022-12-07 18:45:01 [INFO] socket listening on  [plugin: socket]
2022-12-07 18:45:01 [INFO] metric server at : 192.168.24.30:8081 [plugin: Prometheus]
```

NOW the error is because the sg-core core for tcp sockets hasn't merged yet; but this PR is good to go.

**Review by leifmadsen — APPROVED** [review]
I've read the code and I've read through the testing scenario overview. This all looks good to me.

**Pull request merged**